libraries/chain/block_log.cpp

```diff
@@ -185,6 +185,16 @@ namespace eosio { namespace chain {
          return block;
       }
 
+      template <typename Stream>
+      void read_serialized_block(Stream&& ds, uint64_t block_size, std::vector<char>& buff) {
+         assert(buff.empty());
+
+         try {
+            buff.resize(block_size);
+            ds.read(buff.data(), block_size); // append to buff
+         } catch (...) { buff.clear(); } // restore buff to its original state
```

Comment:
With this `catch` and the assert might as well just return the vector as it is the same thing with return value optimization.

Reply:
Will revert back to return vector.